rosbag2_tests/test/rosbag2_tests/test_rosbag2_record_end_to_end.py

```diff
@@ -0,0 +1,87 @@
+# Copyright 2020 Open Source Robotics Foundation, Inc
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+import re
+import time
+import shutil
+import unittest
+
+import launch
+import launch.actions
+import launch_testing
+import launch_testing.actions
+import launch_testing.asserts
+import launch_testing.markers
+
+import pytest
+
+
+@pytest.mark.launch_test
+@launch_testing.markers.keep_alive
+def generate_test_description():
+    # This is necessary to get unbuffered output from the process under test
+    proc_env = os.environ.copy()
+    proc_env['PYTHONUNBUFFERED'] = '1'
+
+    ros2_topic1_pub_process = launch.actions.ExecuteProcess(
+        cmd=['ros2', 'topic', 'pub', '/rosbag2_record_topic1', 'test_msgs/msg/Strings', '{string_value: rosbag2 test message 1}'],
+        output='screen', env=proc_env,
+    )
+    ros2_topic2_pub_process = launch.actions.ExecuteProcess(
+        cmd=['ros2', 'topic', 'pub', '/rosbag2_record_topic2', 'test_msgs/msg/Strings', '{string_value: rosbag2 test message 2}'],
+        output='screen', env=proc_env,
+    )
+
+    return launch.LaunchDescription([
+        ros2_topic1_pub_process,
+        ros2_topic2_pub_process,
+        launch_testing.actions.ReadyToTest()
+    ]), locals()
+
+
+class TestRecord(unittest.TestCase):
+
+    def test_ros2_bag_record(self, launch_service, proc_info, proc_output, proc_env):
+        """Test terminating_proc without command line arguments."""
+        print('Running ros2 bag record')
+
+        dir_path = 'rosbag2_tmp_bag'
+        try:
+            shutil.rmtree(dir_path)
+        except OSError as e:
+            print("Error: %s : %s" % (dir_path, e.strerror))
```

Comment:
@Karsten1987 you may want to use [`setUp()`](https://docs.python.org/3/library/unittest.html#unittest.TestCase.setUp) and [`tearDown()`](https://docs.python.org/3/library/unittest.html#unittest.TestCase.tearDown) methods to create and destroy temporary resources.